autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/configuration/AutofillRuntimeConfigProvider.kt

```diff
@@ -0,0 +1,86 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.configuration
+
+import com.duckduckgo.app.email.EmailManager
+import com.duckduckgo.autofill.store.AutofillStore
+import com.duckduckgo.deviceauth.api.DeviceAuthenticator
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import timber.log.Timber
+import javax.inject.Inject
+
+interface AutofillRuntimeConfigProvider {
+    suspend fun getRuntimeConfiguration(
+        rawJs: String,
+        url: String?
+    ): String
+}
+
+@ContributesBinding(AppScope::class)
+class RealAutofillRuntimeConfigProvider @Inject constructor(
+    private val emailManager: EmailManager,
+    private val deviceAuthenticator: DeviceAuthenticator,
+    private val autofillStore: AutofillStore,
+    private val runtimeConfigurationWriter: RuntimeConfigurationWriter
+) : AutofillRuntimeConfigProvider {
+    override suspend fun getRuntimeConfiguration(
+        rawJs: String,
+        url: String?
+    ): String {
+        Timber.v("BrowserAutofill: getRuntimeConfiguration called")
+
+        val contentScope = runtimeConfigurationWriter.generateContentScope()
+        val userUnprotectedDomains = runtimeConfigurationWriter.generateUserUnprotectedDomains()
+        val userPreferences = runtimeConfigurationWriter.generateUserPreferences(
+            autofillCredentials = determineIfAutofillEnabled(),
+            showInlineKeyIcon = true
+        )
+        val availableInputTypes = generateAvailableInputTypes(url)
+
+        return rawJs
+            .replace("// INJECT contentScope HERE", contentScope)
+            .replace("// INJECT userUnprotectedDomains HERE", userUnprotectedDomains)
+            .replace("// INJECT userPreferences HERE", userPreferences)
+            .replace("// INJECT availableInputTypes HERE", availableInputTypes)
+    }
+
+    private suspend fun generateAvailableInputTypes(url: String?): String {
+        val credentialsAvailable = determineIfCredentialsAvailable(url)
+        val emailAvailable = determineIfEmailAvailable()
+
+        val json = runtimeConfigurationWriter.generateResponseGetAvailableInputTypes(credentialsAvailable, emailAvailable).also {
+            Timber.v("availableInputTypes for %s: \n%s", url, it)
+        }
+        return "availableInputTypes = $json"
+    }
+
+    // in the future, we'll also tie this into feature toggles and remote config
+    private fun determineIfAutofillEnabled(): Boolean =
```

Comment:
FYI we are already parsing autofill from the remote config so you should be able to use `FeatureToggle` and `Autofill` APIs here.

Reply:
Thanks @marcosholgado. It's a good FYI. shouldn't be done in this PR to avoid moving lots around AND changing logic in one go. For a follow-up PR 👍